### DISCUSSION TITLE
Combine OAuth2PasswordBearer with APIKeyHeader (Two layers security)

### DISCUSSION THREAD

**Discussion by jpainam** [fastapi/fastapi#9601] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Header, HTTPException, Depends, Security
from fastapi import Depends, HTTPException, status
from fastapi.security import OAuth2PasswordBearer
from fastapi.security.api_key import APIKeyHeader
from jose import jwt
import datetime

api_key_header = APIKeyHeader(name="x-auth-token", auto_error=False)

reusable_oauth2 = OAuth2PasswordBearer(
    tokenUrl="/login/access-token"
)


def get_current_user(
    token: str = Depends(reusable_oauth2)
):
    credentials_exception = HTTPException(
        status_code=status.HTTP_401_UNAUTHORIZED,
        detail="Could not validate credentials",
        headers={"WWW-Authenticate": "Bearer"},
    )
    try:
        payload = jwt.decode(
            token=token,
            key="SECRET_KEY",
            algorithms=["HS256"],
        )  # noqa: E501

        username: str = payload.get("sub")
        if username is None or username != "username":
            raise credentials_exception
    except:
        raise credentials_exception
    return True



def verify_api_key(api_key_header: str = Security(api_key_header)):
    if api_key_header != "secured_api_key":
        raise HTTPException(
            status_code=HTTP_401_UNAUTHORIZED, detail="Unauthorized API-KEY"
        )


app = FastAPI(
    dependencies=[Depends(verify_api_key)],
)

@app.post("/login/access-token")
def login_for_access_token(
    request: Request,
    form_data: OAuth2PasswordRequestForm = Depends(),
) -> Any:  
    """
    OAuth2 compatible token login, get an access token for future requests
    """
    
    if form_data.username != "username" && form_data.password != "mypassword":
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail="Incorrect username or password",
            headers={"WWW-Authenticate": "Bearer"},
        )

    to_encode = {"sub":"username"}
    expire = datetime.utcnow() + timedelta(minutes=15)
    to_encode.update({"exp": expire})
    encoded_jwt = jwt.encode(to_encode, "SECRET_KEY", algorithm="HS256")  # noqa: E501
    return {
      "access_token": encoded_jwt,
      "token_type": "bearer"
    }

@app.get("/")
async def welcome(
  current_user = Depends(get_current_user),
):
    return {"msg": "Welcome"}
```

### Description

When I open `localhost:5254/docs`, it opens the swagger UI, and the authorization forms
<img width="648" alt="Screenshot 2023-06-01 at 9 12 55 AM" src="https://github.com/tiangolo/fastapi/assets/1701705/cb63cb81-b9ef-46e6-8451-207f53e71146">

I filled out the `apiKey` with my `secured_api_key`  and authorized,  but the `OAuth2PasswordRequestForm` fails. I look at the `Request Headers`, and  it turns out `OAuth2PasswordRequestForm` doesn't sentd the `APIKeyHeader` in the post request.  it depends on `APIKeyHeader`,

<img width="886" alt="Screenshot 2023-06-01 at 9 19 39 AM" src="https://github.com/tiangolo/fastapi/assets/1701705/4c409d76-d70e-4574-be5b-05786f345edb">

I was expecting it to send the `apiKey` as all my endpoints will be protected by an `apiKey`, then `OAuth2PasswordRequestForm` is used to authenticate a user.

Using Postman and sending `x-auth-token` in the header works.

How can I secured my endpoints (including `login/acess-token`) with an `APIKeyHeader` and still use `SwaggerUI`?

### Operating System

macOS

### FastAPI Version

0.95.1

### Python Version

3.11.3

**Comment by link89** [comment]
@jpainam Do you figure out a solution for such use case? I am thinking of using oath2password for login session and apikey for sdk

**Comment by jpainam** [comment]
Not yet. Just bypassing `APIKeyHeader` for the moment.

**Comment by YuriiMotov** [comment]
As I understand, this limitation comes from Swagger UI, doesn't it?
So, it's better to ask this on their repository
